src/brushes/PencilBrush.spec.ts

```diff
@@ -0,0 +1,274 @@
+        const e = { target: canvas.upperCanvasEl } as unknown as TPointerEvent;
+        const pointer = canvas.getScenePoint({
+          ...e,
+          clientX: 10,
+          clientY: 10,
+        });
+        const pointer2 = canvas.getScenePoint({
+          ...e,
+          clientX: 15,
+          clientY: 100,
+        });
+        const pointer3 = canvas.getScenePoint({
+          ...e,
+          clientX: 20,
+          clientY: 160,
+        });
+        const pointer4 = canvas.getScenePoint({
+          ...e,
+          clientX: 320,
+          clientY: 100,
+        });
+        const pointer5 = canvas.getScenePoint({
+          ...e,
+          clientX: 100,
+          clientY: 100,
+        });
+        brush.onMouseDown(pointer, { e });
+        brush.onMouseMove(pointer2, { e });
+        brush.onMouseMove(pointer3, { e });
+        brush.onMouseMove(pointer4, { e });
+        brush.onMouseMove(pointer5, { e });
+        // @ts-expect-error -- protected
+        const pathData = brush.convertPointsToSVGPath(brush._points);
+        expect(
+          pathData,
+          'path data create a path that does not go beyond canvas',
+        ).toEqual(
+          parsePath(
+            'M 9.999 9.999 Q 10 10 12.5 55 Q 15 100 57.5 100 L 100.001 100',
+          ),
+        );
+        // @ts-expect-error -- protected
+        expect(brush._points.length, '2 points have been discarded').toBe(4);
+      });
+
+      it('fires events and creates path on mouse up', () => {
+        let fireBeforePathCreatedEvent = false;
+        let firePathCreatedEvent = false;
+        let added = null;
+
+        canvas.on('before:path:created', () => {
+          fireBeforePathCreatedEvent = true;
+        });
+
+        canvas.on('path:created', (opt) => {
+          firePathCreatedEvent = true;
+          added = opt.path;
+        });
+
+        const brush = new PencilBrush(canvas);
+        const e = { target: canvas.upperCanvasEl } as unknown as TPointerEvent;
+        const pointer = canvas.getScenePoint({
+          ...e,
+          clientX: 10,
+          clientY: 10,
+        });
+        const pointer2 = canvas.getScenePoint({
+          ...e,
+          clientX: 15,
+          clientY: 15,
+        });
+        const pointer3 = canvas.getScenePoint({
+          ...e,
+          clientX: 20,
+          clientY: 20,
+        });
+
+        brush.onMouseDown(pointer, { e });
+        brush.onMouseMove(pointer2, { e });
+        brush.onMouseMove(pointer3, { e });
+        brush.onMouseMove(pointer2, { e });
+        brush.onMouseMove(pointer3, { e });
+        brush.onMouseUp({ e });
+
+        expect(
+          fireBeforePathCreatedEvent,
+          'before:path:created event is fired',
+        ).toBe(true);
+        expect(firePathCreatedEvent, 'path:created event is fired').toBe(true);
+        expect(added, 'a path is added').toBeInstanceOf(Path);
+        expect(added!.path.length, 'path has 6 steps').toBe(6);
```

Comment:
@asturur can you have a look at this one?
it is failing because in jsdom it is yielding 4 instead of 6
seems like there is a difference in browser and jsdom for paths
I am not that familiar with paths in canvas so maybe you can figure out what the issue here is

Reply:
I looked into this 30 minutes.
I see the code in decimatePath is killing the point before the last one, skipping to iterate over it.
I added a TODO but for now it puzzles me that it was working with qunit, it should have failed there too.